# Construction of pYPK0_FBA1tp_ScTAL1_PDC1tp (pMEC1130)


[pYPKa_Z_FBA1tp](pYPKa_Z_FBA1tp.ipynb)

[pYPKa_E_PDC1tp](pYPKa_E_PDC1tp.ipynb)

In [1]:
import pydna
from pydna import ipynb_importer

In [2]:
from ypk_std_primers import p166, p167, p567, p568

[Yeast Pathway Kit Standard Primers](ypk_std_primers.ipynb)

In [3]:
from Bio.Restriction import ZraI, AjiI, EcoRV

In [4]:
pYPK0 = pydna.read("pYPK0.gb")

In [5]:
promoter_clone = pYPKa_Z_FBA1tp = pydna.read("pYPKa_Z_FBA1tp.gb")

[Genbank](http://www.ncbi.nlm.nih.gov/nucleotide/NM_001182243.1) record

In [7]:
from IPython.display import HTML
HTML('<iframe src=http://www.ncbi.nlm.nih.gov/pubmed/9169869 width=100% height=250></iframe>')

In [6]:
gb = pydna.Genbank("bjornjobbb@gmail.com")
gene_template = gb.nucleotide("NM_001182243.1")

In [7]:
gene_template.list_features()

+----------+-----------+-------+------+--------+--------------+--------+------+
| Feature# | Direction | Start | End  | Length | id           | type   | orf? |
+----------+-----------+-------+------+--------+--------------+--------+------+
| 0        |    -->    |   0   | 1008 |   1008 | <unknown id> | source | yes  |
| 1        |    -->    |   0   | 1008 |   1008 | <unknown id> | gene   | yes  |
| 2        |    -->    |   0   | 1008 |   1008 | <unknown id> | CDS    | yes  |
+----------+-----------+-------+------+--------+--------------+--------+------+

In [8]:
orf = gene_template.extract_feature(2)

In [9]:
orf.isorf()

True

In [10]:
print len(orf)
orf.seguid()

1008


GgJ9s-lK11rrZmXZDHjE3AjiVrs

In [11]:
print str(orf.seq.translate())

MSEPAQKKQKVANNSLEQLKASGTVVVADTGDFGSIAKFQPQDSTTNPSLILAAAKQPTYAKLIDVAVEYGKKHGKTTEEQVENAVDRLLVEFGKEILKIVPGRVSTEVDARLSFDTQATIEKARHIIKLFEQEGVSKERVLIKIASTWEGIQAAKELEEKDGIHCNLTLLFSFVQAVACAEAQVTLISPFVGRILDWYKSSTGKDYKGEADPGVISVKKIYNYYKKYGYKTIVMGASFRSTDEIKNLAGVDYLTISPALLDKLMNSTEPFPRVLDPVSAKKEAGDKISYISDESKFRFDLNEDAMATEKLSEGIRKFSADIVTLFDLIEKKVTA*


In [12]:
pf, pr = pydna.parse('''>TAL1_YPK_rec_fwd 
                        gtcgaggaacgccaggttgcccactttctcactagtgaAAATGTCTGAACCAGCTCAAAAGAA
                        >TAL1_YPK_rec_rev 
                        ATTTAAatcctgatgcgtttgtctgcacagatggcgcgTTAAGCGGTAACTTTCTTTTCAATCA''')

In [13]:
g = pydna.pcr( pf, pr, orf)

In [14]:
g.figure()

                                        5ATGTCTGAACCAGCTCAAAAGAA...TGATTGAAAAGAAAGTTACCGCTTAA3
                                                                   |||||||||||||||||||||||||| tm 52.1 (dbd) 65.6
                                                                  3ACTAACTTTTCTTTCAATGGCGAATTgcgcggtagacacgtctgtttgcgtagtcctaAATTTA5
5gtcgaggaacgccaggttgcccactttctcactagtgaAAATGTCTGAACCAGCTCAAAAGAA3
                                         ||||||||||||||||||||||| tm 52.6 (dbd) 65.3
                                        3TACAGACTTGGTCGAGTTTTCTT...ACTAACTTTTCTTTCAATGGCGAATT5

In [15]:
terminator_clone = pYPKa_E_PDC1tp = pydna.read("pYPKa_E_PDC1tp.gb")

In [16]:
p  = pydna.pcr( p167, p567, promoter_clone)
t  = pydna.pcr( p568, p166, terminator_clone)

In [17]:
pYPK0_E_Z, stuffer = pYPK0.cut((EcoRV, ZraI))

In [18]:
(pYPK0_E_Z, p, g, t)

(Dseqrecord(-5681), Amplicon(792), Amplicon(1086), Amplicon(1061))

In [19]:
asm = pydna.Assembly((pYPK0_E_Z, p, g, t), limit=31)

In [20]:
asm

Assembly:
Sequences........................: [5681] [792] [1086] [1061]
Sequences with shared homologies.: [5681] [792] [1061] [1086]
Homology limit (bp)..............: 31
Number of overlaps...............: 4
Nodes in graph(incl. 5' & 3')....: 6
Only terminal overlaps...........: No
Circular products................: [8374]
Linear products..................: [8472] [8435] [8424] [8411] [7730] [7411] [7375] [6681] [6375] [2852] [2110] [1828] [98] [61] [50] [37]

In [21]:
candidate = asm.circular_products[0]
candidate.figure()

 -|pYKP0|98
|        \/
|        /\
|        98|792bp_PCR_prod|38
|                          \/
|                          /\
|                          38|1086bp_PCR_prod|35
|                                             \/
|                                             /\
|                                             35|1061bp_PCR_prod|61
|                                                                \/
|                                                                /\
|                                                                61-
|                                                                   |
 -------------------------------------------------------------------

In [25]:
pYPK0_FBA1tp_ScTAL1_PDC1tp = candidate.synced(pYPK0)

The pYPK0_FBA1tp_ScTAL1_PDC1tp has cseguid ```z56XvyucVzpo2iNPYy5TbXbxWlk``` and 8374 bp.

In [26]:
print len(pYPK0_FBA1tp_ScTAL1_PDC1tp)
pYPK0_FBA1tp_ScTAL1_PDC1tp.cseguid()

8374


z56XvyucVzpo2iNPYy5TbXbxWlk

In [27]:
pYPK0_FBA1tp_ScTAL1_PDC1tp.name="pMEC1130"
#pYPK0_FBA1tp_ScTAL1_PDC1tp.description = "pYPK0_FBA1tp_ScTAL1_PDC1tp"
pYPK0_FBA1tp_ScTAL1_PDC1tp.stamp()

cSEGUID_z56XvyucVzpo2iNPYy5TbXbxWlk_2015-05-09T15:53:49.086533

In [32]:
pYPK0_FBA1tp_ScTAL1_PDC1tp.write("pYPK0_FBA1tp_ScTAL1_PDC1tp.gb")

##Download sequence

[pYPK0_FBA1tp_ScTAL1_PDC1tp](pYPK0_FBA1tp_ScTAL1_PDC1tp.gb)

In [ ]:
import pydna

In [33]:
reloaded = pydna.read("pYPK0_FBA1tp_ScTAL1_PDC1tp.gb")

In [34]:
reloaded.cseguid()

z56XvyucVzpo2iNPYy5TbXbxWlk

In [35]:
reloaded.description

'pYPK0_FBA1tp_ScTAL1_PDC1tp cSEGUID_z56XvyucVzpo2iNPYy5TbXbxWlk_2015-05-09T15:53:49.086533'